src/osal/lv_freertos.c

```diff
@@ -193,17 +188,28 @@ lv_result_t lv_thread_sync_wait(lv_thread_sync_t * pxCond)
 {
     lv_result_t lvRes = LV_RESULT_OK;
 
+    /* If the cond is uninitialized, perform initialization. */
+    prvCheckCondInit(pxCond);
+
 #if USE_FREERTOS_TASK_NOTIFY
-    LV_UNUSED(pxCond);
+    TaskHandle_t current_task_handle = xTaskGetCurrentTaskHandle();
```

Comment:
```
while(!pxCond->xSyncSignal) {
    ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
}
pxCond->xSyncSignal = pdFALSE;
````

Reply:
I'm not completely clear on this. Is this a different solution?

Reply:
I mean to replace all the code under `#if USE_FREERTOS_TASK_NOTIFY` with the above suggestion and see if it works to you.